python_ta/contracts/__init__.py

```diff
@@ -210,8 +210,12 @@ def new_setattr(self: klass, name: str, value: Any) -> None:
             original_attr_value = super(klass, self).__getattribute__(name)
         super(klass, self).__setattr__(name, value)
         frame_locals = inspect.currentframe().f_back.f_locals
-        if self is not frame_locals.get("self"):
+        caller_self = frame_locals.get("self")
+        if not isinstance(caller_self, type(self)):
             # Only validating if the attribute is not being set in a instance/class method
+            # AND self is an instance of caller_self's type -- aka caller_self is equal to
```

Comment:
The check for `isinstance(caller_self, type(self))` is good (it's fine for an instance of a child class to violate an RI on another instance of the a parent class).

In the comment, the wording should be reversed (caller_self is an instance of self's type); the part including and after the `--` is not necessary and can be removed.

Reply:
Thanks for following up on the question!